src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheckTest.java

```diff
@@ -403,4 +411,53 @@ public void testReceiverParameter() throws Exception {
         verify(checkConfig, getPath("InputHiddenFieldReceiver.java"), expected);
     }
 
+    /**
+     * We cannot reproduce situation when visitToken is called and leaveToken is not.
+     * So, we have to use reflection to be sure that even in such situation
+     * state of the field will be cleared.
+     *
+     * @throws Exception when code tested throws exception
+     */
+    @Test
+    public void testClearState() throws Exception {
+        final HiddenFieldCheck check = new HiddenFieldCheck();
+        final Optional<DetailAST> classDef = TestUtil.findTokenInAstByPredicate(
+            JavaParser.parseFile(
+                new File(getPath("InputHiddenField.java")),
+                JavaParser.Options.WITHOUT_COMMENTS),
+            ast -> ast.getType() == TokenTypes.CLASS_DEF);
```

Comment:
please move `JavaParser.parseFile` out to some local variable. It is hard to read this code.